Libplanet/Net/Swarm.cs

```diff
@@ -436,6 +432,77 @@ IEnumerator IEnumerable.GetEnumerator()
             }
         }
 
+        private async Task ProcessMessageAsync<T>(
+            Blockchain<T> blockchain,
+            Message message,
+            CancellationToken cancellationToken)
+            where T : IAction
+        {
+            switch (message)
+            {
+                case Ping ping:
+                    _logger.Debug($"Ping received.");
+                    var reply = new Pong
+                    {
+                        Identity = ping.Identity,
+                    };
+                    await ReplyAsync(reply, cancellationToken);
+                    break;
+
+                case Messages.PeerSetDelta peerSetDelta:
+                    _deltas.Enqueue(peerSetDelta.Delta);
+                    break;
+
+                case GetBlocks getBlocks:
+                    IEnumerable<HashDigest<SHA256>> inventories =
+                        blockchain.FindNextHashes(
+                            getBlocks.Locator, getBlocks.Stop, 500);
+
+                    var inv = new Inventory(inventories)
+                    {
+                        Identity = getBlocks.Identity,
+                    };
+                    await ReplyAsync(inv, cancellationToken);
+                    break;
+
+                case GetData getData:
+                    await TransferBlocks(
+                        blockchain, getData, cancellationToken);
+                    break;
+
+                default:
+                    Trace.Fail($"Can't handle message. [{message}]");
+                    break;
+            }
+        }
+
+        private async Task ReplyAsync(Message message, CancellationToken token)
+        {
+            NetMQMessage netMQMessage = message.ToNetMQMessage(_privateKey);
+            await _router.SendMultipartMessageAsync(
+                netMQMessage,
+                cancellationToken: token);
+        }
+
+        private async Task TransferBlocks<T>(
+            Blockchain<T> blockchain,
+            GetData getData,
+            CancellationToken cancellationToken)
+            where T : IAction
+        {
+            foreach (HashDigest<SHA256> hash in getData.BlockHashes)
+            {
+                if (blockchain.Blocks.TryGetValue(hash, out Block<T> block))
+                {
+                    Message response = new Block(block.ToBencodex(true, true))
```

Comment:
The `Block` that is being instantiated here is `Message.Block` not `Blocks.Block`, right?

Reply:
Yes, its' `Libplanet.Net.Messgeas.Block` (without generic parameter `T`).

Reply:
👌